### PR TITLE
Add the start of basic tests for MRC rotation

### PR DESCRIPTION
**Pull request by jsturtevant**
Signed-off-by: James Sturtevant <jstur@microsoft.com>

**Description**:
Add the start of basic tests for MRC rotation:
- runs some basic tests with MRC turned on
- Starts testing the Rotation of Certificates for Tressor, CertManager, and Vault
~- Fixes a bug in the validator selection logic (see https://github.com/kubernetes/kubernetes/issues/92157#issuecomment-825160958)~ (fixed in https://github.com/openservicemesh/osm/pull/5191)
- upgrades vault version to 1.11 which has new Root Cert Rotation APIs (https://learn.hashicorp.com/tutorials/vault/pki-engine#step-7-rotate-root-ca)

**Testing done**:

e2e

**Affected area**:
| Functional Area            |     |
| -------------------------- | --- |
| New Functionality          | [ ] |
| CI System                  | [ ] |
| CLI Tool                   | [ ] |
| Certificate Management     | [x] |
| Control Plane              | [ ] |
| Demo                       | [ ] |
| Documentation              | [ ] |
| Egress                     | [ ] |
| Ingress                    | [ ] |
| Install                    | [ ] |
| Networking                 | [ ] |
| Observability              | [ ] |
| Performance                | [ ] |
| SMI Policy                 | [ ] |
| Security                   | [ ] |
| Sidecar Injection          | [ ] |
| Tests                      | [x] |
| Upgrade                    | [ ] |
| Other                      | [ ] |

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project?
    -   Did you notify the maintainers and provide attribution?
    no

2. Is this a breaking change?
no

3. Has documentation corresponding to this change been updated in the [osm-docs](https://github.com/openservicemesh/osm-docs) repo (if applicable)?
n/a

### CODE REVIEW

**Comment by jsturtevant** [comment]
the e2e test is timeout with the pod coming up: #5139.

**Comment by jsturtevant** [comment]
This partially implements https://github.com/openservicemesh/osm/issues/4835
- OSM Install (MRC successfully created) 
- Manual Root Certificate Rotation Happy Path for each provider

Each PR that implements part of the rotation process can add the additional checks to the e2e tests to verify the various stages and states of the certificates.

**Review by keithmattix — APPROVED** [review]

**Review by jaellio — APPROVED** [review]

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -241,7 +241,7 @@ jobs:
       matrix:
         k8s_version: [""]
         focus: [""]
-        bucket: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+        bucket: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]
```
[the remark this answers is not in the document]
Comment (jsturtevant) [reply .github/workflows/main.yml:244]: It seems like there were two or three tests per bucket to keep things fast (I assume). When I ran too many tests at a time locally, I also ran into a timeout from go test (default is 10 mins).

File: tests/e2e/e2e_meshrootcertificate_test.go [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+	_, err = createMeshRootCertificate(newCertName, PassiveIntent, installOpts.CertManager)
+	Expect(err).NotTo(HaveOccurred())
+
+	// no secrets are created in Vault case
+	if installOpts.CertManager != Vault {
+		By("ensuring the new CA secret exists")
+		err = Td.WaitForCABundleSecret(Td.OsmNamespace, newCertName, time.Second*90)
+		Expect(err).NotTo(HaveOccurred())
+	}
+	// todo add checks for the correct statuses for the two certificates and complete cert rotation
```
[the remark this answers is not in the document]
Comment (jsturtevant) [reply tests/e2e/e2e_meshrootcertificate_test.go]: updated

File: .github/workflows/main.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (keithmattix) [reply .github/workflows/main.yml:244]: Hmm, I wonder if some of our test flakes are from not having enough buckets...

**Comment by jsturtevant** [comment]
Test failed that time, it looks like the osm bootstrap is crashlooping:


```
Oct 04 18:12:17 osm-e2e-worker kubelet[281]: E1004 18:12:17.761120     281 pod_workers.go:951] "Error syncing pod, skipping" err="failed to \"StartContainer\" for \"osm-bootstrap\" with CrashLoopBackOff: \"back-off 40s restarting failed container=osm-bootstrap pod=osm-bootstrap-59898f89d9-r8kxw_osm-system(411e7da1-5932-41a5-be88-7b3089216a0d)\"" pod="osm-system/osm-bootstrap-59898f89d9-r8kxw" podUID=411e7da1-5932-41a5-be88-7b3089216a0d
Oct 04 18:12:18 osm-e2e-worker kubelet[281]: I1004 18:12:18.972638     281 scope.go:110] "RemoveContainer" containerID="981f66ff4b327feaa05d1b302025c1235f76ad035d6ae5ad8556a87901aee43e"
Oct 04 18:12:18 osm-e2e-worker kubelet[281]: E1004 18:12:18.972947     281 pod_workers.go:951] "Error syncing pod, skipping" err="failed to \"StartContainer\" for \"osm-bootstrap\" with CrashLoopBackOff: \"back-off 40s restarting failed container=osm-bootstrap pod=osm-bootstrap-59898f89d9-r8kxw_osm-system(411e7da1-5932-41a5-be88-7b3089216a0d)\"" pod="osm-system/osm-bootstrap-59898f89d9-r8kxw" podUID=411e7da1-5932-41a5-be88-7b3089216a0d
Oct 04 18:12:32 osm-e2e-worker kubelet[281]: I1004 18:12:32.152468     281 scope.go:110] "RemoveContainer" containerID="981f66ff4b327feaa05d1b302025c1235f76ad035d6ae5ad8556a87901aee43e"
Oct 04 18:12:32 osm-e2e-worker kubelet[281]: E1004 18:12:32.152789     281 pod_workers.go:951] "Error syncing pod, skipping" err="failed to \"StartContainer\" for \"osm-bootstrap\" with CrashLoopBackOff: \"back-off 40s restarting failed container=osm-bootstrap pod=osm-bootstrap-59898f89d9-r8kxw_osm-system(411e7da1-5932-41a5-be88-7b3089216a0d)\"" pod="osm-system/osm-bootstrap-59898f89d9-r8kxw" podUID=411e7da1-5932-41a5-be88-7b3089216a0d
Oct 04 18:12:44 osm-e2e-worker kubelet[281]: I1004 18:12:44.152784     281 scope.go:110] "RemoveContainer" containerID="981f66ff4b327feaa05d1b302025c1235f76ad035d6ae5ad8556a87901aee43e"
Oct 04 18:12:44 osm-e2e-worker kubelet[281]: E1004 18:12:44.153095     281 pod_workers.go:951] "Error syncing pod, skipping" err="failed to \"StartContainer\" for \"osm-bootstrap\" with CrashLoopBackOff: \"back-off 40s restarting failed container=osm-bootstrap pod=osm-bootstrap-59898f89d9-r8kxw_osm-system(411e7da1-5932-41a5-be88-7b3089216a0d)\"" pod="osm-system/osm-bootstrap-59898f89d9-r8kxw" podUID=411e7da1-5932-41a5-be88-7b3089216a0d
Oct 04 18:12:59 osm-e2e-worker kubelet[281]: I1004 18:12:59.154886     281 scope.go:110] "RemoveContainer" containerID="981f66ff4b327feaa05d1b302025c1235f76ad035d6ae5ad8556a87901aee43e"
```

and the logs for that pod are failing to call the osm-control-plane-validator?

```
2022-10-04T18:12:17.084528866Z stderr F {"level":"fatal","component":"osm-bootstrap","error":"Internal error occurred: failed calling webhook \"osm-control-plane-validator.k8s.io\": failed to call webhook: Post \"https://osm-validator.osm-system.svc:9093/validate?timeout=10s\": dial tcp 10.96.153.210:9093: connect: connection refused","time":"2022-10-04T18:12:17Z","file":"osm-bootstrap.go:171","message":"Error setting up default MeshRootCertificate osm-mesh-root-certificate from ConfigMap preset-mesh-root-certificate"}
```

**Comment by jsturtevant** [comment]
This seems like it is leaving around a validating webhook registration, i've seen this when testing locally and deleting things manually from e2e cluster without cleaning it up

**Comment by jsturtevant** [comment]
It looks like in this failure case the cluster is re-used.  Maybe there is a race when cleaning up and removing the webhooks?

File: .github/workflows/main.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jsturtevant) [reply .github/workflows/main.yml:244]: I think I found at least one of the flakes: https://github.com/openservicemesh/osm/pull/5182#issuecomment-1267565020

**Comment by jsturtevant** [comment]
I reproduced this locally and the validating webhook is left over by the test `Upgrade from latest`:

```
 k get validatingwebhookconfigurations.admissionregistration.k8s.io -A          NAME                     WEBHOOKS   AGE                                          osm-validator-mesh-osm   2          3m33s          
```

The reason this is flaking is dependent on the order the test are run.  In a successful test run, the SPIFFE test runs first: https://github.com/openservicemesh/osm/actions/runs/3184127777/jobs/5192238528

In the failed test run  `Upgrade from latest` runs first leaving the validation webhook left over which causes the failure with timeout to the validation webhook (it doesn't exist until later in the startup process. 

I wasn't able to reproduce this on a branch without this change so it may be due to validator selection?  Looking into the upgrade test next.

**Comment by jsturtevant** [comment]
Yes the change to 

```
			Key:      "kubernetes.io/metadata.name",
							Operator: metav1.LabelSelectorOpIn,
							Values:   []string{osmNamespace},
```

is causing this because the validating webhook is left over from the `Upgrade from latest` test.  Looking into why that gets left behind

**Comment by jsturtevant** [comment]
found it, the cleanup logic does it for current version its running under: 

https://github.com/openservicemesh/osm/blob/main/charts/osm/templates/cleanup-hook.yaml#L89

In this case the webhook version was  `app.kubernetes.io/version: v1.2.1` but after upgrade the version that the cleanuphook was trying to delete is `app.kubernetes.io/version: latest-main`

**Comment by jsturtevant** [comment]
It looks like we work around this in the osm cli with the `--delete-cluster-wide-resources` (which isn't used on the test): https://github.com/openservicemesh/osm/blob/main/cmd/cli/uninstall_mesh.go#L390-L396

**Comment by jsturtevant** [comment]
This commit should fix it https://github.com/openservicemesh/osm/pull/5182/commits/1a2140787a08bf252ce1835a4875762b38ea7cdb

I believe this is a bug and will split these changes out to a separate PR.

**Comment by jsturtevant** [comment]
I've split this out into #5191, will rebase after that merges

File: .github/workflows/main.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jsturtevant) [reply .github/workflows/main.yml:244]: This flake I found doesn't seem to be the randomness seen sometimes in #5139, This was specific to the change here and fixed in #5191

**Comment by jsturtevant** [comment]
> I've split this out into #5191, will rebase after that merges

rebased with those changes

**Pull request merged**